### ISSUE TITLE
[a11y]: (WebComponents) CDS-Select - There is no way to visually hide the label while maintaining accesiblity

### ISSUE THREAD

**Issue by olihf-dematic** [carbon-design-system/carbon#20649]
### Package

@carbon/web-components

### Browser

Chrome

### Operating System

Windows

### Package version

2.27.1

### React version

N/A

### Automated testing tool and ruleset

Chrome Lighthouse

### Description

When `label-text` is provided, and `hide-label` is true. The following Lighthouse error is reported:

 `Select elements do not have associated label elements.`

There doesn't seem to be any workaround to apply an aria-label to the html select component, which forces the user to render a label or have this error.

Possible solution: When hide-label is true, use label-text as aria-label (if provided).

### WCAG 2.1 Violation

WCAG 2.1 (A), WCAG 2.0 (A), WCAG 2.2 (A), Section 508, Trusted Tester, EN 301 549

### Reproduction/example

https://stackblitz.com/edit/github-xfkunee4-qfyqlqf8?file=index.html

### Steps to reproduce

Set `label-text` to 'Any value'
Set `hide-label` tor true

Scan component

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by olihf-dematic** [comment]
I'm happy to work on this

**Comment by heloiselui** [comment]
Hey @olihf-dematic, if you still want to work on this, feel free to open a PR. Just a quick reminder: if a PR isn’t opened within two weeks, the issue becomes available for someone else to take on. You can find more info [here](https://github.com/carbon-design-system/carbon/blob/main/.github/CONTRIBUTING.md). Thanks for helping out!
